File: test/Course_test.java
import org.junit.jupiter.api.Test;

import java.util.ArrayList;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class Course_test {
    @Test
    public void calculateCourseScore_test() {
        CourseInformation courseInformation = new CourseInformation("test", "test");
        int midterm_score = 85;
        int final_score = 85;
        TakenCourse course = new TakenCourse(courseInformation, midterm_score, final_score);
        Score course_score = course.get_course_score();

        assertEquals(course_score, Score.BA, "The scores should be the same!");
    }

    @Test
    public void isStudentQualified_test() {
        DataManagement dataManagement = new DataManagement();
        Student student = dataManagement.getStudent("duyguates@marun.edu.tr");;
        Course course = dataManagement.getCourse("General Biology");
        CourseRequirements courseRequirements = course.getCourseRequirements();
        boolean[] trueAns = {true, true, true, true};
        boolean[] calculatedAns = courseRequirements.isStudentQualified(student);
        for(int i =0; i < trueAns.length; i++) {
            assertEquals(trueAns[i], calculatedAns[i], "Student is not qualified!");
        }
    }

    // we couldn't test checkPrerequisiteCourse method because it is a private method. But isStudentQualified method includes that method

}

File: Test/Course_test.py
import unittest

from pythonCode.Course import CourseInformation, TakenCourse, Score
from pythonCode.DataManagement import DataManagement


class Course_test(unittest.TestCase):
    def test_isStudentQualified_test(self):
        data_management = DataManagement()
        student = data_management.getStudent("selinpolat@marun.edu.tr")
        course = data_management.getCourse("Data Structures")
        courseRequirements = course.getCourseRequirements()
        trueAns = [True,True,True,True]
        calculatedAns = courseRequirements.isStudentQualified(student)
        for i in range(len(trueAns)):
            self.assertEqual(trueAns[i], calculatedAns[i], "Student is not qualified!")

    def  test_calculateCourseScore_test(self):
        courseInformation = CourseInformation("Introduction to Computer Engineering","CS1200")
        midterm_score = 100
        final_score = 95
        course = TakenCourse(courseInformation, midterm_score, final_score)
        course_score = course.getCourseScore()

        self.assertEqual(course_score, Score.AA, "Course score is not calculated correctly!")

